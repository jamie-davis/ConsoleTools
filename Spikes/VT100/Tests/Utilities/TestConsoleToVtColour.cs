using System;
using System.Linq;
using TestConsoleLib;
using TestConsoleLib.Testing;
using VT100.FullScreen;
using VT100.Utilities;
using Xunit;

namespace VT100.Tests.Utilities
{
    public class TestConsoleToVtColour
    {
        [Fact]
        public void AllConsoleColoursAreConverted()
        {
            //Arrange
            var consoleColours = typeof(ConsoleColor).GetFields()
                .Where(f => f.IsLiteral)
                .Select(c => new { Colour = c.Name, Value = (int?)c.GetRawConstantValue() ?? -1 });
            
            //Act
            var result = consoleColours
                .Select(c => new 
                    { 
                        ConsoleColor = c.Colour, 
                        VtColour = ConsoleToVtColour.Convert((ConsoleColor)c.Value) 
                    });
            
            //Assert
            var output = new Output();
            output.FormatTable(result);
            output.Report.Verify();
        }
    }
}