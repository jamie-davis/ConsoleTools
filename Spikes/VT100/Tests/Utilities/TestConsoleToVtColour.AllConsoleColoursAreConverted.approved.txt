Console                                                                                        
Color           Vt Colour                                                                      
----------- -------------                                                                      
Black               Black                                                                      
DarkBlue             Blue                                                                      
DarkGreen           Green                                                                      
DarkCyan             Cyan                                                                      
DarkRed               Red                                                                      
DarkMagenta       Magenta                                                                      
DarkYellow         Yellow                                                                      
Gray                White                                                                      
DarkGray      BrightBlack                                                                      
Blue           BrightBlue                                                                      
Green         BrightGreen                                                                      
Cyan           BrightCyan                                                                      
Red             BrightRed                                                                      
Magenta     BrightMagenta                                                                      
Yellow       BrightYellow                                                                      
White         BrightWhite                                                                      
                                                                                               